### PR TITLE
Fix #51 URLs will be clickable now 

### LINKED ISSUE #51: Make URLs in chat blocks clickable
For message blocks where there is any valid URL, it should be converted to a clickable URL.
## Current Behaviour
Screenshot from 2019-04-23 03-23-36
## Expected Behaviour
Screenshot from 2019-04-23 03-25-00

### PR DESCRIPTION
**Pull request by uptoratlen**
Hi,
as promised the issue #51 is addressed in this pr.
The cr utilizes a copy of  [linkifyjs](https://github.com/Soapbox/linkifyjs), to extract of a string a url in any kind of format.
Two libs are copied, following the existing jMinEmoji2.min.js lib, to static/js... . In addition I copied the license of the linkifyjs.
Link samples are added to the already existing sample chat whatsapp_chat_en.txt in some formats.

Feel free to discuss or change according to your need.
cu

### CODE REVIEW

File: templates/index.html [hunk 1]
```diff
@@ -177,6 +177,8 @@ <h4 class="text-center">
 <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js" integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6" crossorigin="anonymous"></script>
 <script type="text/javascript" src="{{ url_for('static', filename='js/jMinEmoji2.min.js') }}"></script>
 <script type="text/javascript" src="{{ url_for('static', filename='js/script.js') }}"></script>
+<script type="text/javascript" src="{{ url_for('static', filename='js/linkifyjs/linkify.min.js') }}"></script>
+<script type="text/javascript" src="{{ url_for('static', filename='js/linkifyjs/linkify-jquery.min.js') }}"></script>
```
Comment (prabhakar267) [inline templates/index.html:181]: There is already `jquery 3.4.1` present, can you please verify if this is necessary or not?
We should only use one jquery.

File: templates/index.html [hunk 2]
```diff
@@ -177,6 +177,8 @@ <h4 class="text-center">
 <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js" integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6" crossorigin="anonymous"></script>
 <script type="text/javascript" src="{{ url_for('static', filename='js/jMinEmoji2.min.js') }}"></script>
 <script type="text/javascript" src="{{ url_for('static', filename='js/script.js') }}"></script>
+<script type="text/javascript" src="{{ url_for('static', filename='js/linkifyjs/linkify.min.js') }}"></script>
```
Comment (prabhakar267) [inline templates/index.html:180]: Is there a CDN version available for this?

**Review by prabhakar267 — CHANGES_REQUESTED** [review]

File: templates/index.html (hunk 2 above)
Comment (uptoratlen) [reply templates/index.html:180]:
```suggestion
<script src="https://cdnjs.cloudflare.com/ajax/libs/jQuery-linkify/2.1.9/linkify.min.js" integrity="sha512-kxj7VjlzsQgiku2vbRcZI0FJ0dXmPsiRLugiRxJrCROusKHaFfX/hGDD1/L/R0Y+xI8zlA2B5nm6USapz7nQbg==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
```
What do you think? Version 2.1.9 is also in the PR.

File: templates/index.html (hunk 1 above)
Comment (uptoratlen) [reply templates/index.html:181]: Not 100% sure, but a test with out the second (new) jquery works fine. So I think we coudl remove this
```suggestion

```

**Comment by prabhakar267** [comment]
@uptoratlen Can you update the code with the suggested changes? I will be able to merge it then

**Comment by uptoratlen** [comment]
Hi, I committed the changes and also removed the unneeded linkify files (as CDN version is used). 
Cheers

**Review by prabhakar267 — APPROVED** [review]

**Pull request merged**